src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (size == 0) {
+            addFirst(value);
+        } else if (index == 0) {
+            head = new Node(value, head, null);
+            head.next.prev = head;
+            size++;
+        } else if (index == size) {
+            addLast(value);
+        } else {
+            Node current = getNodeByIndex(index);
+            Node newNode = new Node(value, current, current.prev);
+            current.prev.next = newNode;
+            current.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNodeByIndex(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node current = getNodeByIndex(index);
+        T oldElment = current.element;
+        current.element = value;
+        return oldElment;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node current = getNodeByIndex(index);
```

Comment:
Here you use checkIndex method and after this getByIndex, combine it in one method